utils.cpp

```diff
@@ -453,7 +453,8 @@ size_t ggml_quantize_q4_0(float * src, void * dst, int n, int k, int qk, int64_t
 
     assert(k % qk == 0);
 
-    uint8_t pp[qk/2];
+    std::vector<uint8_t> pp;
+    pp.reserve(qk/2);
```

Comment:
Let's use `alloca` as suggested in #36 , I'm worried `std::vector` will make heap allocations. Not terrible since this is the quantization tool, but still.

You probably need the following includes added to utils.cpp to make `alloca` available:

https://github.com/ggerganov/llama.cpp/blob/6b2cb6302ffaf8264e33af1dc52e3ea54003e690/ggml.c#L3-L8

In the quantization routines, do not `ifdef` as suggested in #36, but straight use `alloca` always